pkg/pgmodel/view_test.go

```diff
@@ -0,0 +1,174 @@
+package pgmodel
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/jackc/pgx/v4/pgxpool"
+
+	_ "github.com/jackc/pgx/v4/stdlib"
+	"github.com/prometheus/prometheus/prompb"
+)
+
+func getViewRowCount(t *testing.T, db *pgxpool.Pool, view string, expected int) {
+	var count int
+	err := db.QueryRow(context.Background(), fmt.Sprintf("SELECT count(*) FROM %s", view)).Scan(&count)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if count != expected {
+		t.Fatalf("unexpected view count: view %s, got %d expected %d", view, count, expected)
+	}
+}
+
+func TestSQLView(t *testing.T) {
+	if testing.Short() {
+		t.Skip("skipping integration test")
+	}
+
+	withDB(t, *database, func(db *pgxpool.Pool, t *testing.T) {
+		metrics := []prompb.TimeSeries{
+			{
+				Labels: []prompb.Label{
+					{Name: metricNameLabelName, Value: "no tags"},
+				},
+				Samples: []prompb.Sample{
+					{Timestamp: 10, Value: 0.1},
+				},
+			},
+			{
+				Labels: []prompb.Label{
+					{Name: metricNameLabelName, Value: "reserved tags"},
+					{Name: "foo", Value: "bar"},
+					{Name: "labels", Value: "val1"},
+					{Name: "series_id", Value: "vaL2"},
+					{Name: "time", Value: "val1"},
+					{Name: "value", Value: "val3"},
+				},
+				Samples: []prompb.Sample{
+					{Timestamp: 10, Value: 0.1},
+					{Timestamp: 40, Value: 0.4},
+				},
+			},
+			{
+				Labels: []prompb.Label{
+					{Name: metricNameLabelName, Value: "long tags names"},
+					{Name: "foo", Value: "bar"},
```

Comment:
Seeing the below maybe you do have something to deal with long names, perhaps its in another bit of the code that I'm missing?